dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java

```diff
@@ -825,9 +814,10 @@ public void setForceHandleUse(boolean v)
      */
     public int getRemainingBufferCapacity()
     {
-        int totalBufferCapacity = this.readBufferCapacity.get();
-        int currentBufferLen = this.readBufferDataLen.get();
-        return totalBufferCapacity - currentBufferLen;
+        synchronized (readBuffer)
+        {
+            return readBuffer.getSpace();
```

Comment:
it feels like renaming to getAvailableSpace or remainingBufferCapacity would be slightly better for maintenance